File: python/claude/p82.py
"""
p82.py
Simple Flask app with /api route that takes a JSON payload and returns the 'name' field.
Run: pip install flask
"""
from flask import Flask, request, jsonify
import json

app = Flask(__name__)

@app.route('/api', methods=['GET', 'POST'])
def api():
    if request.method == 'GET':
        payload = request.args.get('payload')
    else:
        payload = request.form.get('payload') or request.get_json()
    
    if not payload:
        return jsonify({'error': 'missing payload'}), 400
    
    try:
        if isinstance(payload, str):
            data = json.loads(payload)
        else:
            data = payload
        name = data.get('name')
        return jsonify({'name': name})
    except Exception as e:
        return jsonify({'error': str(e)}), 400

if __name__ == '__main__':
    app.run(port=5000)

File: javascript/claude/p82.js
// p82.js - Express /api endpoint with JSON payload
// Run: npm install express
const express = require('express');
const app = express();
app.use(express.json());

app.get('/api', (req, res) => {
  const payload = req.query.payload;
  if (!payload) return res.status(400).json({ error: 'missing payload' });
  try {
    const data = JSON.parse(payload);
    return res.json({ name: data.name });
  } catch (e) {
    return res.status(400).json({ error: e.message });
  }
});

app.post('/api', (req, res) => {
  if (!req.body) return res.status(400).json({ error: 'missing payload' });
  return res.json({ name: req.body.name });
});

app.listen(5000, () => console.log('p82.js listening on 5000'));
